### ISSUE TITLE
Not passed test

### ISSUE THREAD

**Issue by GitScrum** [eslint/eslint#7799]
Not passed test `cli when supplied with report output file path should write the file and create dirs if they don't exist:`

**Tell us about your environment**

* **ESLint Version: 3.12.2
* **Node Version: 7.2.0
* **npm Version: 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using? Default

**What did you do? Please include the actual source code causing the issue.**

```bash
$ npm test
```

**What did you expect to happen?**

all passed tests

**What actually happened? Please include the actual, raw output from ESLint.**

```sh
  13566 passing (1m)
  1 failing

  1) cli when supplied with report output file path should write the file and create dirs if they don't exist:
     AssertionError: expected '\nC:\\Users\\ivdemidov\\AppData\\Local\\Temp\\eslint\\fixtures\\single-quoted.js\n  1:13  warning  Strings must use doublequote  quotes\n\n✖ 1 problem (0 errors, 1 warning)\n' to include 'C:\\Users\\IVDEMI~1\\AppData\\Local\\Temp\\eslint\\fixtures\\single-quoted.js'
      at Function.assert.include (D:\sand-box\git\eslint\node_modules\chai\lib\chai\interface\assert.js:843:45)
      at Context.it (D:\sand-box\git\eslint\tests\lib\cli.js:541:20)
      at callFn (D:\sand-box\git\eslint\node_modules\mocha\lib\runnable.js:326:21)
      at Test.Runnable.run (D:\sand-box\git\eslint\node_modules\mocha\lib\runnable.js:319:7)
      at Runner.runTest (D:\sand-box\git\eslint\node_modules\mocha\lib\runner.js:422:10)
      at D:\sand-box\git\eslint\node_modules\mocha\lib\runner.js:528:12
      at next (D:\sand-box\git\eslint\node_modules\mocha\lib\runner.js:342:14)
      at D:\sand-box\git\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (D:\sand-box\git\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate.<anonymous> (D:\sand-box\git\eslint\node_modules\mocha\lib\runner.js:320:5)



=============================================================================
Writing coverage object [D:\sand-box\git\eslint\coverage\coverage.json]
Writing coverage reports at [D:\sand-box\git\eslint\coverage]
=============================================================================

=============================== Coverage summary ===============================
Statements   : 99.48% ( 11563/11624 ), 147 ignored
Branches     : 98.2% ( 8891/9054 ), 75 ignored
Functions    : 99.9% ( 1999/2001 ), 22 ignored
Lines        : 99.47% ( 11490/11551 )
================================================================================

START:
20 12 2016 16:17:43.628:WARN [karma]: Port 9876 in use
20 12 2016 16:17:43.631:INFO [karma]: Karma v0.13.22 server started at http://localhost:9877/
20 12 2016 16:17:43.636:INFO [launcher]: Starting browser PhantomJS
20 12 2016 16:17:45.261:INFO [PhantomJS 2.1.1 (Windows 8 0.0.0)]: Connected on socket ep7LPmVfCAehOM8-AAAA with id 88475412

Finished in 0.644 secs / 0.589 secs

SUMMARY:
√ 233 tests completed
npm ERR! Test failed.  See above for more details.
```

**Comment by platinumazure** [comment]
Not just me then! Excellent. I can reproduce this in my machine and I have Node 7.x as well. (I only started seeing this issue when I upgraded my Node version.)

**Comment by gyandeeps** [comment]
Wasn't able to reproduce this on 
* Windows 10
* Node - 7.2.0

**Comment by platinumazure** [comment]
@gyandeeps It also seems to depend on one's directory structure. In both OP's and my case, our ESLint project is in a user folder and our usernames are both longer than 8 characters. It's probably situational and has to do with how Windows normalizes path names.

**Comment by not-an-aardvark** [comment]
Does the issue still occur if you change [this line](https://github.com/eslint/eslint/blob/aafebb2958416b825283bc428d64bd96b3adf63b/tests/lib/cli.js#L78) to use `path.resolve.apply` instead of `path.join.apply`?

**Comment by platinumazure** [comment]
@not-an-aardvark I'll give that a try at some point in the next few days. Thanks for the suggestion!

**Comment by kaicataldo** [comment]
@platinumazure Any luck with this?

**Comment by platinumazure** [comment]
Finally took a look yesterday, couldn't find an obvious fix by using
path.normalize or path.resolve. Frustratingly, when you search for "node",
"path", and "tilde", you get Unix path expansion questions, even when you
add "Windows". I'm definitely still seeing the test failure and only on
Node 7, but I haven't made any real headway beyond that.

**Comment by kaicataldo** [comment]
@platinumazure Thanks for the update :+1:

**Comment by mysticatea** [comment]
I had not reproduced this on my shell (GitHub Shell of [GitHub Desktop for Windows](https://desktop.github.com/)) for long time. But today, I found this happens on `cmd.exe`.
(My project path is `C:\Users\t-nagashima.AD\Documents\GitHub\eslint`)

As a workaround for now, maybe you can use GitHub Shell :)

**Comment by mysticatea** [comment]
I'm not sure why, but it reproduced with `mocha .\tests\lib\*.js --reporter progress`, but it did not reproduce with `mocha .\tests\lib\cli.js --reporter progress`.

**Comment by mysticatea** [comment]
It reproduced with `mocha .\tests\lib\{cli,config}.js --reporter progress`

**Comment by mysticatea** [comment]
It seems to be fixed if I remove `require("mock-fs")` from `tests/lib/config.js`.
